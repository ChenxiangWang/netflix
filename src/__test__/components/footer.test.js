import React from "react";
import { render, screen } from "@testing-library/react";
import Footer from "../../components/footer";
describe("<Footer>", () => {
  // it = test(name, fn, timeout);
  it("render the <Footer> with a populated data", () => {
    const { container } = render(
      <Footer>
        <Footer.Title>Question? Contact us</Footer.Title>
        <Footer.Break />
        <Footer.Row>
          <Footer.Colunm>
            <Footer.Link href="#">FAQ</Footer.Link>
            <Footer.Link href="#">Investor Relations</Footer.Link>
            <Footer.Link href="#">Ways to watch</Footer.Link>
            <Footer.Link href="#">Coporate Information</Footer.Link>
            <Footer.Link href="#">Netflix Originals</Footer.Link>
          </Footer.Colunm>

          <Footer.Colunm>
            <Footer.Link href="#">Help Center</Footer.Link>
            <Footer.Link href="#">Jobs</Footer.Link>
            <Footer.Link href="#">Terms of Use</Footer.Link>
            <Footer.Link href="#">Contact us</Footer.Link>
          </Footer.Colunm>

          <Footer.Colunm>
            <Footer.Link href="#">Account</Footer.Link>
            <Footer.Link href="#">Redeem Gift Cards</Footer.Link>
            <Footer.Link href="#">Privacy</Footer.Link>
            <Footer.Link href="#">Speed Test</Footer.Link>
          </Footer.Colunm>

          <Footer.Colunm>
            <Footer.Link href="#">Media Center</Footer.Link>
            <Footer.Link href="#">Buy Gift Cards</Footer.Link>
            <Footer.Link href="#">Cookie Preferences</Footer.Link>
            <Footer.Link href="#">Legal Notices</Footer.Link>
          </Footer.Colunm>
        </Footer.Row>
        <Footer.Break />
        <Footer.Text>Netflix United Kingdom</Footer.Text>
      </Footer>
    );
    expect(screen.getAllByText("Question? Contact us")).toBeTruthy();
    expect(screen.getAllByText("FAQ")).toBeTruthy();
    expect(screen.getAllByText("Help Center")).toBeTruthy();
    expect(screen.getAllByText("Account")).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
