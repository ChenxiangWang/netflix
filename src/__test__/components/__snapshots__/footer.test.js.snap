// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer> render the <Footer> with a populated data 1`] = `
<div
  class="sc-jSMfEi kRkSeI"
>
  <div
    class="sc-eCYdqJ iWzuUk"
  >
    Question? Contact us
  </div>
  <div
    class="sc-gKXOVf hRegEz"
  />
  <div
    class="sc-bczRLJ igRvSZ"
  >
    <div
      class="sc-gsnTZi fGmbmf"
    >
      <a
        class="sc-hKMtZM jRWTXB"
      >
        FAQ
      </a>
      <a
        class="sc-hKMtZM jRWTXB"
      >
        Investor Relations
      </a>
      <a
        class="sc-hKMtZM jRWTXB"
      >
        Ways to watch
      </a>
      <a
        class="sc-hKMtZM jRWTXB"
      >
        Coporate Information
      </a>
      <a
        class="sc-hKMtZM jRWTXB"
      >
        Netflix Originals
      </a>
    </div>
    <div
      class="sc-gsnTZi fGmbmf"
    >
      <a
        class="sc-hKMtZM jRWTXB"
      >
        Help Center
      </a>
      <a
        class="sc-hKMtZM jRWTXB"
      >
        Jobs
      </a>
      <a
        class="sc-hKMtZM jRWTXB"
      >
        Terms of Use
      </a>
      <a
        class="sc-hKMtZM jRWTXB"
      >
        Contact us
      </a>
    </div>
    <div
      class="sc-gsnTZi fGmbmf"
    >
      <a
        class="sc-hKMtZM jRWTXB"
      >
        Account
      </a>
      <a
        class="sc-hKMtZM jRWTXB"
      >
        Redeem Gift Cards
      </a>
      <a
        class="sc-hKMtZM jRWTXB"
      >
        Privacy
      </a>
      <a
        class="sc-hKMtZM jRWTXB"
      >
        Speed Test
      </a>
    </div>
    <div
      class="sc-gsnTZi fGmbmf"
    >
      <a
        class="sc-hKMtZM jRWTXB"
      >
        Media Center
      </a>
      <a
        class="sc-hKMtZM jRWTXB"
      >
        Buy Gift Cards
      </a>
      <a
        class="sc-hKMtZM jRWTXB"
      >
        Cookie Preferences
      </a>
      <a
        class="sc-hKMtZM jRWTXB"
      >
        Legal Notices
      </a>
    </div>
  </div>
  <div
    class="sc-gKXOVf hRegEz"
  />
  <p
    class="sc-dkzDqf bunCUP"
  >
    Netflix United Kingdom
  </p>
</div>
`;
